homeassistant/components/xiaomi_ble/sensor.py

```diff
@@ -68,6 +69,55 @@
         native_unit_of_measurement=LIGHT_LUX,
         state_class=SensorStateClass.MEASUREMENT,
     ),
+    # Impedance sensor (ohm)
+    (DeviceClass.IMPEDANCE, Units.OHM): SensorEntityDescription(
+        key=f"{DeviceClass.IMPEDANCE}_{Units.OHM}",
+        icon="mdi:omega",
+        native_unit_of_measurement=Units.OHM,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # Mass sensor (kg)
+    (DeviceClass.MASS, Units.MASS_KILOGRAMS): SensorEntityDescription(
+        key=f"{DeviceClass.MASS}_{Units.MASS_KILOGRAMS}",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.KILOGRAMS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # Mass sensor (lb)
+    (DeviceClass.MASS, Units.MASS_POUNDS): SensorEntityDescription(
+        key=f"{DeviceClass.MASS}_{Units.MASS_POUNDS}",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.POUNDS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # Mass non stabilized sensor (kg)
+    (DeviceClass.MASS_NON_STABILIZED, Units.MASS_KILOGRAMS): SensorEntityDescription(
+        key=f"{DeviceClass.MASS_NON_STABILIZED}_{Units.MASS_KILOGRAMS}",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.KILOGRAMS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # Mass non stabilized sensor (lb)
+    (DeviceClass.MASS_NON_STABILIZED, Units.MASS_POUNDS): SensorEntityDescription(
+        key=f"{DeviceClass.MASS_NON_STABILIZED}_{Units.MASS_POUNDS}",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.POUNDS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # Mass stabilized sensor (kg)
+    (DeviceClass.MASS_STABILIZED, Units.MASS_KILOGRAMS): SensorEntityDescription(
+        key=f"{DeviceClass.MASS_STABILIZED}_{Units.MASS_KILOGRAMS}",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.KILOGRAMS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # Mass stabilized sensor (lb)
+    (DeviceClass.MASS_STABILIZED, Units.MASS_POUNDS): SensorEntityDescription(
+        key=f"{DeviceClass.MASS_STABILIZED}_{Units.MASS_POUNDS}",
+        device_class=SensorDeviceClass.WEIGHT,
+        native_unit_of_measurement=UnitOfMass.POUNDS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
```

Comment:
I don't think we should add all those sensors:
- There's no reason to add duplicated sensors with native units in kg and lb, Home Assistant automatically converts. The exception would be if a scale sold in the US only makes the lb sensors available, and the scales sold elsewhere only makes the kg sensors available
- What is the difference between "mass", "non stabilized mass" and "stabilized mass"? Are all three needed? Should two of them be given the entity category `EntityCategory.DIAGNOSTIC` or be hidden by default?
- How are the sensors named?

Reply:
- A MiScale is broadcasting the weight in kg, if it is set to kg and in lbs if it is set to lbs. A workaround can be that we convert lbs measurements to kg in xiaomi-ble, and always forward it in kg to HA. 
- non-stabilized mass is the weight when you step on the scale and the weight is still varying. At a certain moment, the weight is stabilized (constant) on the scale, after which the scale sends a BLE message that the scale is stabilized, with the actual weight. I think EntityCategory.DIAGNOSTIC for the non-stabilized mass is a good suggestion, most users won't be interested in that number after all
- Sensors are names `mass stabilized` `mass non stabilized`. I propose I make the following changes: Use `mass` instead of `mass stabilized` and use `mass non stabilized` as diagnostic sensor.